Forte.EpiServer.AzureSearch/ContentExtractor/ContentExtractorController.cs

```diff
@@ -0,0 +1,82 @@
+using System.Collections.Generic;
+using System.Linq;
+using EPiServer.Core;
+using Forte.EpiServer.AzureSearch.Extensions;
+using Forte.EpiServer.AzureSearch.Model;
+
+namespace Forte.EpiServer.AzureSearch.ContentExtractor
+{
+    public class ContentExtractorController : IContentExtractorController
+    {
+        public const string BlockExtractedTextFragmentsSeparator = " ";
+        private readonly IEnumerable<IContentExtractor> _extractors;
+
+        public ContentExtractorController(IEnumerable<IContentExtractor> contentExtractors)
+        {
+            _extractors = contentExtractors;
+        }
+
+        public IEnumerable<string> Extract(IContent content)
+        {
+            return ExtractInternal(content, new HashSet<ContentReference>(), this)
+                .SelectMany(r => r.Values)
+                .Where(v => string.IsNullOrEmpty(v) == false)
+                .Distinct();
+        }
+        
+        public IEnumerable<string> Extract(IContentData content)
+        {
+            return ExtractInternal(content, this)
+                .SelectMany(r => r.Values)
+                .Where(v => string.IsNullOrEmpty(v) == false)
+                .Distinct();
+        }
+
+        private IEnumerable<ContentExtractionResult> ExtractInternal(IContent content,
+            ISet<ContentReference> visitedContent, ContentExtractorController extractor)
+        {
+            var result = new List<ContentExtractionResult>();
+
+            if (visitedContent.Contains(content.ContentLink))
+            {
+                return result;
+            }
+
+            visitedContent.Add(content.ContentLink);
+
+            if (content.IsDeleted)
+            {
+                return Enumerable.Empty<ContentExtractionResult>();
```

Comment:
Why sometimes you return
`Enumerable.Empty<ContentExtractionResult>();`
and sometimes `return result;`(a few lines above)?